sensor/debugger/k8s/fake.go

```diff
@@ -318,9 +400,48 @@ func (f *FakeEventsManager) createEvent(msg resources.InformerK8sMsg, ch chan<-
 }
 
 // waitOnMode waits depending on the mode
-func (f *FakeEventsManager) waitOnMode() {
+func (f *FakeEventsManager) waitOnMode(events []string) error {
 	switch f.Mode {
 	case Delay:
 		time.Sleep(f.Delay)
+		return nil
+	case ChannelAck:
+		if len(events) == 0 {
+			return nil
+		}
+		receivedEvents := 0
+		for {
+			timeout := time.After(5 * time.Second)
+			select {
+			case <-timeout:
+				return errors.New("timeout reached waiting for event")
+			case event, more := <-f.AckChannel:
+				if !more {
+					return errors.New("channel closed")
+				}
+				for _, e := range events {
+					sensorEvent := &central.SensorEvent{}
+					if err := jsonpb.UnmarshalString(e, sensorEvent); err != nil {
+						return fmt.Errorf("error unmarshaling '%s'", e)
+					}
+					if sensorEvent.GetResource() == nil {
+						return fmt.Errorf("resource not found in sensor event '%s'", e)
+					}
+					resource := reflect.ValueOf(sensorEvent.GetResource())
+					compareFunc, ok := sensorEventCompareFunctions[resource.Type().String()]
+					if !ok {
+						return fmt.Errorf("compare function for resource '%s' not found", resource.Type().String())
+					}
+					if compareFunc(sensorEvent.GetResource(), event) {
+						receivedEvents++
+						break
+					}
+				}
+				if receivedEvents == len(events) {
+					return nil
+				}
```

Comment:
It looks like a good candidate for refactoring into a separate func.

Reply:
Done